### PR TITLE
NGraph python api cannot be built w/o onnx importer

### PR DESCRIPTION
**Pull request by slyubimt**
### Details:
 - fix link dependencies for ngraph python module 

### Tickets:
 - 51338

### CODE REVIEW

File: ngraph/python/CMakeLists.txt [hunk 1]
```diff
@@ -85,7 +85,8 @@ file(GLOB_RECURSE SOURCES src/pyngraph/*.cpp)
 pybind11_add_module(_${PROJECT_NAME} MODULE ${SOURCES})
 
 target_include_directories(_${PROJECT_NAME} PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src")
-target_link_libraries(_${PROJECT_NAME} PRIVATE ngraph::ngraph ngraph::onnx_importer)
+target_link_libraries(_${PROJECT_NAME} PRIVATE ngraph::ngraph)
+add_dependencies(_${PROJECT_NAME} ngraph::onnx_importer)
```
Comment (tomdol) [inline ngraph/python/CMakeLists.txt:89]: This doesn't look right. When building WITH onnx_importer, you're not linking with it any more. Will this work?

Comment (ilya-lavrenov) [reply ngraph/python/CMakeLists.txt:89]: BTW, anyway, it does not solve the original issue. The general idea is to be able to build ngraph python api with `-DNGRAPH_ONNX_IMPORT_ENABLE=OFF`

Comment (ilya-lavrenov) [reply ngraph/python/CMakeLists.txt:89]: Ok, with such change I have:
```
-- Configuring done
CMake Warning (dev) at ngraph/python/CMakeLists.txt:90 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "ngraph::onnx_importer" of target "_pyngraph" does
  not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
```

Maybe we need:
```cmake
if (TARGET ngraph::onnx_importer)
    add_dependencies(_${PROJECT_NAME} ngraph::onnx_importer)
endif()
```
In this case we will not see any errors and ONNX importer becomes an optional dependency.

Comment (tomdol) [reply ngraph/python/CMakeLists.txt:89]: This looks like the proper solution to the original problem. I'm just wondering if we should link to the onnx importer inside the same `if`. OTOH the "OpenVino ONNX CI" passed and onnx importer is used via the python bindings there so perhaps we don't need it.

Comment (slyubimt) [reply ngraph/python/CMakeLists.txt]: Initially onnx_importer library was a part of ngraph wheel package, but was added to link dependencies by mistake after refactoring. This PR fixes that.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
are you OK to merge?

**Review by jiwaszki — APPROVED** [review]

**Comment by ilyachur** [comment]
@tomdol Please have a look

**Review by tomdol — APPROVED** [review]

**Pull request merged**